lnpbp-0002.md

```diff
@@ -112,29 +112,27 @@ from a given Bitcoin script, based on Miniscript [16].
 ### Commitment procedure
 
 The **committing party**, having a message `msg`, must:
-1. Collect all unique public key instances (named **original public key set** 
-   `P`) related to a given transaction output, defined as:
-   - a single public key for P2PK, P2PKH, P2WPK, P2WPK-in-P2SH type of outputs;
-   - an arbitrary public key (even with an unknown corresponding private key),
-     if `OP_RETURN` type of `scriptPubkey` is used;
-   - an intermediate public key for the Taproot-based (SegWit v1) output;
-   - all public keys extracted using 
-     [algorithm](#deterministic-public-key-extraction-from-bitcoin-script) from 
-     either:
-     * `witnessScript` contained in the `witness` stack for native v0 P2WSH
-       and legacy P2WSH-in-P2SH SegWit outputs;
-     * `redeemScript` contained in top  `scriptSig` for non-SegWit P2SH
-     * or bare (custom) script-based `scriptPubkey`
-2. Select a single public key `Po` from the set `P` of the original public keys 
-   for the tweaking procedure. It is advised that the corresponding private key
-   being controlled by the committing party, which will simplify future spending
-   of the output.
-3. Run [LNPBP-1 commitment procedure](https://github.com/LNP-BP/LNPBPs/blob/master/lnpbp-0001.md#commitment-procedure)
-   for the message `msg`, selected public key `Po`, public key set `P` and a
-   protocol-specific `tag`, provided by the upstream protocol using this 
-   standard.
+1. Collect all public keys instances (named **original public key set**) 
+   related to the given transaction output, defined as:
+   - a single public key for P2PK, P2PKH, P2WPK, P2WPK/WSH-in-P2SH type of outputs;
+   - an arbitrary public key (even with an unknown corresponding private key), 
+     if `OP_RETURN` `scriptPubkey` is used;
```

Comment:
I think we need to clarify `OP_RETURN` handing options here:
```suggestion
   - If `OP_RETURN` `scriptPubkey` is used, it must contain a single public key serialized in BIP-340 xcoordonly form right after `OP_RETURN` opcode; for all other forms of `OP_RETURN` data algorithm must fail;
```

And add to the rationale:
```text
Use of `OP_RETURN`-based commitments is not advised, but if it can't be avoided (for instance because of HSM use) it requires the public key containing commitment to be serialized in BIP-340 format such that the data will look like any other 256-bit hash, common for `OP_RETURN` outputs, increasing anonymity set and preventing advanced chain analysis.
```